starknet_devnet/util.py

```diff
@@ -15,16 +15,20 @@
     StorageEntry,
 )
 from starkware.starknet.testing.contract import StarknetContract
-from starkware.starkware_utils.error_handling import StarkException
+from starkware.starkware_utils.error_handling import StarkErrorCode, StarkException
 
 
-def custom_int(arg: str) -> int:
+def validate_hex_string(arg: str) -> int:
     """
-    Converts the argument to an integer.
-    Conversion base is 16 if `arg` starts with `0x`, otherwise `10`.
+    Converts the argument to an integer only if it starts with `0x`.
     """
-    base = 16 if arg.startswith("0x") else 10
-    return int(arg, base)
+    if arg.startswith("0x"):
+        return int(arg, 16)
+
+    raise StarknetDevnetException(
```

Comment:
I see this error being defined in multiple places. Why not just derive a new class from `StarknetDevnetException` with the code and message set in advance?

Reply:
Let's discuss it on the call